resources/todomvc/architecture-examples/preact/src/todo/components/input.jsx

```diff
@@ -0,0 +1,58 @@
+import { useCallback, useRef, useEffect } from "preact/hooks";
+
+const sanitize = (string) => {
+    const map = {
+        "&": "&amp;",
+        "<": "&lt;",
+        ">": "&gt;",
+        '"': "&quot;",
+        "'": "&#x27;",
+        "/": "&#x2F;",
+    };
+    const reg = /[&<>"'/]/gi;
+    return string.replace(reg, (match) => map[match]);
+};
+
+const hasValidMin = (value, min) => {
+    return value.length >= min;
+};
+
+export function Input({ onSubmit, placeholder, label, defaultValue, onBlur }) {
+    const inputRef = useRef(null);
+
+    useEffect(() => {
+        if (inputRef.current) {
+            const end = inputRef.current.value.length;
+            inputRef.current.setSelectionRange(end, end);
+            inputRef.current.focus();
+        }
+    }, [inputRef.current])
+
+    const handleBlur = useCallback(() => {
+        if (onBlur)
+            onBlur();
+    }, [onBlur]);
+
+    const handleKeyDown = useCallback(
+        (e) => {
+            if (e.keyCode === 13) {
```

Comment:
nit: I would suggest to use `e.key === "Enter"` instead of `keyCode` that is deprecated.

Another option would be to use a `form` and attach the event handler to the form's `submit` event. My understanding is that this is better supported by assistive technologies, but I might be wrong on that :-)

Reply:
It's probably not a big deal for the benchmark itself, all these events should fire synchronously together.